ansi-x963-kdf/tests/tests.rs

```diff
@@ -0,0 +1,245 @@
+use digest::{Digest, FixedOutputReset};
+use hex_literal::hex;
+use sha2::{Sha224, Sha256, Sha512};
+
+struct Fixture<'a> {
+    secret: &'a [u8],
+    other_info: &'a [u8],
+    expected_key: &'a [u8],
+}
+
+fn test_key_derivation<D>(fixtures: &[Fixture])
+where
+    D: Digest + FixedOutputReset,
+{
+    for Fixture {
+        secret,
+        other_info,
+        expected_key,
+    } in fixtures.iter()
+    {
+        for key_length in 1..expected_key.len() {
+            let mut key = vec![0u8; key_length];
+            assert!(ansi_x963_kdf::derive_key_into::<D>(secret, other_info, &mut key).is_ok());
+            eprintln!("{}", expected_key.len());
+            assert_eq!(&expected_key[..key_length], &key);
+        }
+    }
+}
+
+#[test]
+fn test_input_output_sha224() {
+    let fixtures = [
+        Fixture {
+            secret: &hex!("00"),
+            other_info: &[],
+            expected_key: &hex!(
+                "4a6ebc83b8e2b19eea640500be6bcffdddaa07b8b2f81f2c533940e4e6ad6cfd
```

Comment:
Please link provenance of the used test vectors.

Reply:
As I mentioned, I generated these myself using a Java-based implementation - with a few python spot checks. Is that acceptable?
Otherwise I just saw that pyca/cryptography has some test vectors and the license is probably compatible (both LICENSE.APACHE and LICENSE.BSD)? Would that work better maybe?
Let me know what you prefer, thanks.

Reply:
Yeah, just link the used Java library and write that the vectors were generated by it.

Reply:
Thanks, addressed in 7053d879a43c4be34d5b5ab73cc8c5637fab41fe